exec/java-exec/src/main/java/org/apache/drill/exec/expr/fn/DrillFuncHolder.java

```diff
@@ -264,7 +265,24 @@ public boolean isFieldReader(int i) {
     return this.parameters[i].isFieldReader;
   }
 
-  public MajorType getReturnType(List<LogicalExpression> args) {
+  public static List<MajorType> getResultType(List<LogicalExpression> expressions) {
+    final List<MajorType> argumentTypes = Lists.newArrayList();
+    for (LogicalExpression expression : expressions) {
+      argumentTypes.add(expression.getMajorType());
+    }
+    return argumentTypes;
+  }
+
+  public final MajorType getReturnTypeMajorType(final List<MajorType> types) {
+    final List<LogicalExpression> expressions = Lists.newArrayList();
+    for(final MajorType type : types) {
+      final LogicalExpression dumbExpr = new DumbLogicalExpression(type);
```

Comment:
In stead of creating a DumbLogicalExpression augmented with type, it's better to do the opposite way.  Slightly modify the original one such that it accepts a list of MajorTypes. It only requires to know the argument type, to decide the return type. 

MajorType getReturnTypeMajorType(final List<MajorType> types) : This one uses most part of original implementation.

MajorType getReturnType(List<LogicalExpression> expressions) : this one will pass in the list of argument types and then call the first one.